### PR TITLE
feat: Update deploy contracts

### PR DESCRIPTION
**Pull request by Cphilo**

### CODE REVIEW

File: scripts/deploy_contracts.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  }
+}
+
+async function deployContracts() {
+  // write all contract address to a local file
+
+  let contracts = new Map<string, string>();
+  // deprecated contract, but need set this param.
+  // Need setStakingProxy later.
+  const deprecatedStakingProxy = "0xde12723B2a478D4A1d347D49fe15ac0C2691f2Da";
```
Comment (merlincdj) [inline scripts/deploy_contracts.ts:48]: `const deprecatedStakingProxy = "0xde12723B2a478D4A1d347D49fe15ac0C2691f2Da";` -> `const nonContractAddress = "0x0000000000000000000000000000000000000001";`

File: scripts/deploy_contracts.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  const depAddressList = constants.depAddressList;
+  let args = [depAddressList["woo"], depAddressList["wooPP"], deprecatedStakingProxy];
```
Comment (merlincdj) [inline scripts/deploy_contracts.ts:50]: `deprecatedStakingProxy` -> `nonContractAddress`

File: scripts/setup_contracts.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  const booster = await ethers.getContractAt(
+    "RewardBooster", addressList["RewardBooster"]);
+  const wooStakingCompounder = await ethers.getContractAt(
+    "WooStakingCompounder", addressList["WooStakingCompounder"]);
+  const wooStakingController = await ethers.getContractAt(
+      "WooStakingController", addressList["WooStakingController"]);
+  const wooStakingProxy = await ethers.getContractAt(
+    "WooStakingController", addressList["WooStakingProxy"]);
+
+  await stakingManager.setStakingProxy(mpRewarder.address);
```
Comment (merlincdj) [inline scripts/setup_contracts.ts:89]: `mpRewarder.address` -> `wooStakingProxy.address`

File: scripts/setup_contracts.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  const wethRewarder = await ethers.getContractAt(
+    "SimpleRewarder", addressList["WethRewarder"]);
+  const booster = await ethers.getContractAt(
+    "RewardBooster", addressList["RewardBooster"]);
+  const wooStakingCompounder = await ethers.getContractAt(
+    "WooStakingCompounder", addressList["WooStakingCompounder"]);
+  const wooStakingController = await ethers.getContractAt(
+      "WooStakingController", addressList["WooStakingController"]);
+  const wooStakingProxy = await ethers.getContractAt(
+    "WooStakingController", addressList["WooStakingProxy"]);
```
Comment (merlincdj) [inline scripts/setup_contracts.ts:87]: `WooStakingController` -> `WooStakingProxy`

**Review by merlincdj — APPROVED** [review]

**Comment by Cphilo** [comment]
Resolved comments

**Pull request merged**